platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/model/WiringModel.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring.model;
+
+import com.swirlds.base.state.Startable;
+import com.swirlds.base.state.Stoppable;
+import com.swirlds.base.time.Time;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.wiring.builders.TaskSchedulerBuilder;
+import com.swirlds.common.wiring.builders.TaskSchedulerMetricsBuilder;
+import com.swirlds.common.wiring.builders.TaskSchedulerType;
+import com.swirlds.common.wiring.model.internal.StandardWiringModel;
+import com.swirlds.common.wiring.wires.output.OutputWire;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Set;
+
+/**
+ * A wiring model is a collection of task schedulers and the wires connecting them. It can be used to analyze the wiring
+ * of a system and to generate diagrams.
+ */
+public interface WiringModel extends Startable, Stoppable {
+
+    /**
+     * Build a new wiring model instance.
+     *
+     * @param platformContext the platform context
+     * @param time            provides wall clock time
+     * @return a new wiring model instance
+     */
+    @NonNull
+    static WiringModel create(@NonNull final PlatformContext platformContext, @NonNull final Time time) {
+        return new StandardWiringModel(platformContext, time);
+    }
+
+    /**
+     * Get a new task scheduler builder.
+     *
+     * @param name the name of the task scheduler. Used for metrics and debugging. Must be unique. Must only contain
+     *             alphanumeric characters and underscores.
+     * @return a new wire builder
+     */
+    @NonNull
+    <O> TaskSchedulerBuilder<O> schedulerBuilder(@NonNull final String name);
```

Comment:
Missing javadoc for generic `O`, not sure if we strictly require docs for generics though

Reply:
Added. I think generic javadocs are a good idea to require.